<a href="https://colab.research.google.com/github/crpytoscooby/College/blob/master/Riya_K_(9063)_ML_1A(air).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Machine Learning Lab Assignment 1 (Group-A)**

**Riya Karambelkar**

**TE IT**

**Roll No 9063**

**Air Data**

Importing numpy and pandas libraries

In [2]:
import numpy as np
import pandas as pd

Loading the air_data.csv dataset in the data variable



In [14]:
data=pd.read_csv("/content/air_data.csv - air_data.csv.csv")

# Performing operations on the given dataset:

Printing the first five rows

In [15]:
data.head()

Unnamed: 0  Ozone  Solar.R  Wind  Temp  Month  Day
0           1   41.0    190.0   7.4    67      5    1
1           2   36.0    118.0   8.0    72      5    2
2           3   12.0    149.0  12.6    74      5    3
3           4   18.0    313.0  11.5    62      5    4
4           5    NaN      NaN  14.3    56      5    5

Printing last five rows

In [16]:
data.tail()

Unnamed: 0  Ozone  Solar.R  Wind  Temp  Month  Day
103         104   44.0    192.0  11.5    86      8   12
104         105   28.0    273.0  11.5    82      8   13
105         106   65.0    157.0   9.7    80      8   14
106         107    NaN     64.0  11.5    79      8   15
107         108   22.0     71.0  10.3    77      8   16

Printing shape of data

In [18]:
data.shape

(108, 7)

This means there are 108 rows and 7 columns

Printing the information of dataframe

In [19]:
data.describe()

Unnamed: 0       Ozone     Solar.R  ...        Temp       Month        Day
count   108.00000   74.000000  101.000000  ...  108.000000  108.000000  108.00000
mean     54.50000   42.418919  191.108911  ...   77.398148    6.296296   14.75000
std      31.32092   32.242237   95.527787  ...    9.714562    1.043532    8.80009
min       1.00000    1.000000    7.000000  ...   56.000000    5.000000    1.00000
25%      27.75000   16.500000  101.000000  ...   72.000000    5.000000    7.00000
50%      54.50000   34.500000  220.000000  ...   80.000000    6.000000   14.00000
75%      81.25000   63.750000  273.000000  ...   85.000000    7.000000   22.00000
max     108.00000  135.000000  334.000000  ...   93.000000    8.000000   31.00000

[8 rows x 7 columns]

In [20]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 108 entries, 0 to 107
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  108 non-null    int64  
 1   Ozone       74 non-null     float64
 2   Solar.R     101 non-null    float64
 3   Wind        108 non-null    float64
 4   Temp        108 non-null    int64  
 5   Month       108 non-null    int64  
 6   Day         108 non-null    int64  
dtypes: float64(3), int64(4)
memory usage: 6.0 KB


Finding if there are any missing values in the data

In [21]:
data.isna()

Unnamed: 0  Ozone  Solar.R   Wind   Temp  Month    Day
0         False  False    False  False  False  False  False
1         False  False    False  False  False  False  False
2         False  False    False  False  False  False  False
3         False  False    False  False  False  False  False
4         False   True     True  False  False  False  False
..          ...    ...      ...    ...    ...    ...    ...
103       False  False    False  False  False  False  False
104       False  False    False  False  False  False  False
105       False  False    False  False  False  False  False
106       False   True    False  False  False  False  False
107       False  False    False  False  False  False  False

[108 rows x 7 columns]

In [22]:
print("The total missing values in the dataframe are: ")
print(data.isna().sum())

The total missing values in the dataframe are: 
Unnamed: 0     0
Ozone         34
Solar.R        7
Wind           0
Temp           0
Month          0
Day            0
dtype: int64


Finding datatype of each column

In [23]:
data.dtypes

Unnamed: 0      int64
Ozone         float64
Solar.R       float64
Wind          float64
Temp            int64
Month           int64
Day             int64
dtype: object

Finding out zeros in the dataframe

In [24]:
data.all()

Unnamed: 0    True
Ozone         True
Solar.R       True
Wind          True
Temp          True
Month         True
Day           True
dtype: bool

Hence none of the columns contain zero

Counting number of zeros in each column

In [25]:
(data==0).sum()

Unnamed: 0    0
Ozone         0
Solar.R       0
Wind          0
Temp          0
Month         0
Day           0
dtype: int64

Finding the mean ozone in air

In [26]:
data['Ozone'].mean()

42.41891891891892

Replacing missing values in the Ozone column by the mean ozone


In [28]:
data['Ozone'].fillna(data['Ozone'].mean())

0      41.000000
1      36.000000
2      12.000000
3      18.000000
4      42.418919
         ...    
103    44.000000
104    28.000000
105    65.000000
106    42.418919
107    22.000000
Name: Ozone, Length: 108, dtype: float64

Extracting only Ozone, Wind and Temp colums from the dataset

In [31]:
data1=data[['Ozone', 'Solar.R', 'Wind', 'Temp']]
data1

Ozone  Solar.R  Wind  Temp
0     41.0    190.0   7.4    67
1     36.0    118.0   8.0    72
2     12.0    149.0  12.6    74
3     18.0    313.0  11.5    62
4      NaN      NaN  14.3    56
..     ...      ...   ...   ...
103   44.0    192.0  11.5    86
104   28.0    273.0  11.5    82
105   65.0    157.0   9.7    80
106    NaN     64.0  11.5    79
107   22.0     71.0  10.3    77

[108 rows x 4 columns]

Replacing the missing values with mean values

In [32]:
data1['Solar.R'].mean()

191.1089108910891

In [33]:
data1=data1.fillna(data1.mean())
data1

Ozone     Solar.R  Wind  Temp
0    41.000000  190.000000   7.4    67
1    36.000000  118.000000   8.0    72
2    12.000000  149.000000  12.6    74
3    18.000000  313.000000  11.5    62
4    42.418919  191.108911  14.3    56
..         ...         ...   ...   ...
103  44.000000  192.000000  11.5    86
104  28.000000  273.000000  11.5    82
105  65.000000  157.000000   9.7    80
106  42.418919   64.000000  11.5    79
107  22.000000   71.000000  10.3    77

[108 rows x 4 columns]

Dividing the dataset in training(75%) and testing(25%)

In [34]:
from sklearn.model_selection import train_test_split

train_data, test_data = train_test_split(data1, test_size=0.25)

In [35]:
train_data.shape

(81, 4)

In [36]:
test_data.shape

(27, 4)